.github/workflows/main.yml

```diff
@@ -1,40 +1,72 @@
 ---
 name: CI
 
+env:
+  IMAGE_NAME: research-template
+  PUBLIC_IMAGE_NAME: ghcr.io/opensafely/research-template
+  REGISTRY: ghcr.io
+
 on:
+  workflow_dispatch:
+  pull_request:
   push:
+    branches: [main]
 
 jobs:
-  check:
+  build:
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
+      - uses: hadolint/hadolint-action@54c9adbab1582c2ef04b2016b760714a4bfde3cf  # v3.1.0
+        # Lint the dockerfile before building
+        with:
+          failure-threshold: error
+          ignore: DL3022, DL3008, DL3009
```

Comment:
suggestion: In other repositories, we've added this configuration as a comment in the `Dockerfile`; [example](https://github.com/opensafely-core/job-server/blob/3ba7cfb8cafda4492460b3d3ec876aaebd479c5f/docker/Dockerfile#L8-L11).

I think that's better both:

* for consistency across repositories
* so that hadolint has the same configuration however it's run (for example, locally versus CI)